dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityTypeAttributeStore.java

```diff
@@ -28,13 +28,14 @@
  * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-import org.hisp.dhis.render.type.SectionRenderingObject;
+import org.hisp.dhis.common.IdentifiableObjectStore;
+import org.hisp.dhis.program.ProgramTrackedEntityAttribute;
+import org.hisp.dhis.user.User;
 
-public class SectionRenderTypeDeserialize
-    extends AbstractDeviceRenderTypeMapDeserializer<SectionRenderingObject>
+import java.util.List;
+
+public interface TrackedEntityTypeAttributeStore
+    extends IdentifiableObjectStore<TrackedEntityTypeAttribute>
 {
-    public SectionRenderTypeDeserialize()
-    {
-        super( SectionRenderingObject::new );
-    }
+    List<TrackedEntityAttribute> getAttributesByTrackedEntityTypesWithDataRead( List<TrackedEntityType> trackedEntityTypes, User user );
```

Comment:
1) Please add Javadoc
  2) I would rename this method to getAttributes and explain the ACL part in the java docs

Reply:
Fixed and added unit test